src/Entity/Activity/Registration.php

```diff
@@ -130,18 +138,30 @@ public function setOption(PriceOption $option): self
         return $this;
     }
 
-    public function getPerson(): ?LocalAccount
+    public function getPerson(): ?ContactInterface
     {
-        return $this->person;
+        return $this->person ?? $this->external;
     }
 
-    public function setPerson(?LocalAccount $person): self
+    public function setPerson(?ContactInterface $person): self
     {
-        $this->person = $person;
+        $this->person = null;
+        $this->external = null;
+
+        if ($person instanceof LocalAccount) {
+            $this->person = $person;
+        } elseif ($person instanceof ExternalRegistrant) {
+            $this->external = $person;
+        }
 
         return $this;
     }
 
+    public function isExternal(): bool
```

Comment:
Lovely function. As far as I can see we don't use it, I feel we should leave the creation of function till when we need them.

Reply:
We use it in the templates, specifically in templates/activity/show.html.twig on ln. 82 and ln. 87, as well as in templates/admin/activity/show.html.twig on ln. 207

Reply:
didn't think of looking in the templates. makes a lot of sense to use them there